cmd/server/main.go

```diff
@@ -1,53 +1,148 @@
 package main
 
 import (
+	"bytes"
+	"context"
 	"fmt"
-	"net"
-	"time"
+	"io"
+	"log"
+	"net/http"
+	"runtime"
+	"strings"
 
+	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/onflow/flow-evm-gateway/api"
+	"github.com/onflow/flow-evm-gateway/indexer"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/rs/zerolog"
+)
+
+const (
+	accessURL = "access-001.devnet49.nodes.onflow.org:9000"
 )
 
 func main() {
-	server := api.NewRPCServer()
-	defer server.Stop()
+	store := storage.NewStore()
+	runServer(store)
+	runIndexer(store)
+
+	runtime.Goexit()
+}
+
+func apis(store *storage.Store) []rpc.API {
+	return []rpc.API{
+		{
+			Namespace: api.EthNamespace,
+			Service: &api.BlockChainAPI{
+				Store: store,
+			},
+		},
+	}
+}
+
+func runIndexer(store *storage.Store) {
+	ctx := context.Background()
 
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	chain, err := indexer.GetChain(ctx, accessURL)
 	if err != nil {
-		fmt.Println("can't listen:", err)
+		log.Fatalf("could not get chain: %v", err)
 	}
-	defer listener.Close()
-	go server.ServeListener(listener)
 
+	execClient, err := indexer.NewExecutionDataClient(accessURL, chain)
+	if err != nil {
+		log.Fatalf("could not create execution data client: %v", err)
+	}
+
+	sub, err := execClient.SubscribeEvents(ctx, flow.ZeroID, 0, indexer.EventFilter{
+		Contracts: []string{"A.7e60df042a9c0868.FlowToken"},
+	})
+	if err != nil {
+		log.Fatalf("could not subscribe to execution data: %v", err)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case response, ok := <-sub.Channel():
+			if sub.Err() != nil {
+				log.Fatalf("error in subscription: %v", sub.Err())
+			}
+			if !ok {
+				log.Fatalf("subscription closed")
+			}
```

Comment:
these `log.Fatal` calls are OK initially, but you will eventually want to handle reconnecting gracefully. There's an example of how to do that in the `flow-go-sdk` PR

Reply:
Perfect :ok_hand: 
Added it in this PR as well: https://github.com/onflow/flow-evm-gateway/pull/11/commits/4d47834396bf89b099718b0fac02824827dddc94